examples/http_record_har.rs

```diff
@@ -297,15 +296,15 @@ async fn http_mitm_proxy(req: Request) -> Result<Response, Infallible> {
     // NOTE: in a production proxy you most likely
     // wouldn't want to build this each invocation,
     // but instead have a pre-built one as a struct local
-    let client = EasyHttpWebClient::builder()
+    let client = EasyHttpWebClient::connector_builder()
         .with_default_transport_connector()
         .with_tls_proxy_support_using_boringssl()
         .with_proxy_support()
         .with_tls_support_using_boringssl(Some(Arc::new(base_tls_config)))
         .with_custom_connector(UserAgentEmulateHttpConnectModifierLayer::default())
         .with_default_http_connector()
-        .with_svc_req_inspector(UserAgentEmulateHttpRequestModifier::default())
-        .build();
+        .build_client()
+        .with_jit_layers(UserAgentEmulateHttpRequestModifierLayer::default());
```

Comment:
I think we should rename this to `with_jit_layer`. If you want me to do it I can do it in a follow-up commit after merge, but either way someone will have to rename it. Reason you can give multiple layers as well is because a tuple of layers is also a layer.